### PR TITLE
Generate tchannel server

### PR DESCRIPTION
**Pull request by ChuntaoLu**
This PR is on top of https://github.com/uber/zanzibar/pull/55

### CODE REVIEW

File: examples/example-gateway/build/clients/baz/baz_handler.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
+// Code generated by zanzibar
```
Comment (Raynos) [inline examples/example-gateway/build/clients/baz/baz_handler.go]: This entire file is too much boilerplate to save a few lines. 

Can we just make the tests more boilerplate and delete this ?

File: codegen/templates/tchannel_server.tmpl [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	{{- else}}
+	    r, respHeaders, err := s.handler.{{.Name}}(ctx, reqHeaders, &req)
+	{{- end}}
+
+	{{if eq (len .Exceptions) 0 -}}
+	    if err != nil {
+            return false, nil, nil, err
+	    }
+	    res.Success = r
+    {{else -}}
```
Comment (Raynos) [inline codegen/templates/tchannel_server.tmpl]: whitespace of else is not readable.

File: codegen/templates/tchannel_server.tmpl [hunk 3]
```diff
@@ -0,0 +1,124 @@
+{{- /* template to render edge gateway tchannel server code */ -}}
+// Code generated by zanzibar
+// @generated
+
+// Package {{.PackageName}} is generated code used to make or handle TChannel calls using Thrift.
+package {{.PackageName}}
+
+import (
+	"context"
```
Comment (Raynos) [inline codegen/templates/tchannel_server.tmpl]: mixed spaces and tabs FFFFFFFfff----

we need more linters :(

Comment (ChuntaoLu) [inline codegen/templates/tchannel_server.tmpl]: fix in https://github.com/uber/zanzibar/pull/57/commits/785e981803a15744326f7872135867214e956386

File: codegen/templates/tchannel_server.tmpl (hunk 2 above)
Comment (ChuntaoLu) [inline codegen/templates/tchannel_server.tmpl]: fix in https://github.com/uber/zanzibar/pull/57/commits/785e981803a15744326f7872135867214e956386

File: examples/example-gateway/build/clients/baz/baz_handler.go (hunk 1 above)
Comment (ChuntaoLu) [inline examples/example-gateway/build/clients/baz/baz_handler.go]: With all the changes already in part3, it's easier to work this out over there..., so fix by https://github.com/uber/zanzibar/pull/57/commits/39bfe645dce529a7334439c5d821bff6dea11883 and https://github.com/uber/zanzibar/pull/57/commits/8cb587ea117e6a0a70aa9576b6d9326048785416

**Pull request merged**